### PR TITLE
feature #3009 - restyled chat screen

### LINKED ISSUE #3009: Rework chat UI
### User Story

As a user, I want a clean chat experience.

### Description

Find mockup details [here](https://projects.invisionapp.com/share/KXEWN9XHM#/screens/268975254)

e.g. 

selection_021

selection_022

This issue does not include chat details screen.

*Type*: Feature

### PR DESCRIPTION
**Pull request by goranjovic**
fixes #3009

### Summary:

Changed Chat screen, as per designs.

Omitted any changes related to chat input area, to be submitted in a separate PR by @alwx 

### Steps to test:
- Try direct chat
- Try group chat
- Try public chat

status: ready 

Designs: https://zpl.io/VYWL0kL

### CODE REVIEW

**Comment by goranjovic** [comment]
simulator screen shot feb 7 2018 at 1 59 46 pm
simulator screen shot feb 7 2018 at 2 00 12 pm
simulator screen shot feb 7 2018 at 2 00 21 pm

File: src/status_im/chat/views/message/message.cljs [hunk 1]
```diff
@@ -236,22 +227,31 @@
   (letsubs [{:keys [photo-path]} [:get-current-account]]
     (photo from photo-path)))
 
+(defview message-author-name [from]
+  (letsubs [username    [:contact-name-by-identity from]]
+    [react/text {:style {:font-size      12
```
Comment (jeluard) [inline src/status_im/chat/views/message/message.cljs]: Can we move this to styles? Or maybe some component? ;)

**Review by jeluard — APPROVED** [review]

File: src/status_im/chat/views/message/message.cljs (hunk 1 above)
Comment (alwx) [reply src/status_im/chat/views/message/message.cljs]: agree, it's better to move it to styles first

File: src/status_im/chat/views/toolbar_content.cljs [hunk 2]
```diff
@@ -36,29 +34,37 @@
...
-    (str (label :t/sync-in-progress) " " percentage "% " currentBlock)))
+    (str (i18n/label :t/sync-in-progress) " " percentage "% " currentBlock)))
+
+(defn last-activity-syncing [sync-state]
+  (letsubs [state [:get :sync-data]]
+    [react/text {:style styles.screen/last-activity-text}
+     (case sync-state
+       :in-progress (in-progress-text state)
+       :synced (i18n/label :t/sync-synced)
+       nil)]))
```
Comment (alwx) [inline src/status_im/chat/views/toolbar_content.cljs]: should it be `defview` instead of `defn`?

**Review by alwx — CHANGES_REQUESTED** [review]

File: src/status_im/chat/views/toolbar_content.cljs (hunk 2 above)
Comment (goranjovic) [reply src/status_im/chat/views/toolbar_content.cljs]: Good catch. The really weird thing is that this worked

**Comment by goranjovic** [comment]
@jeluard @alwx updated both items

File: src/status_im/chat/styles/message/message.cljs [hunk 3]
```diff
@@ -45,11 +47,11 @@
   (let [align     (if outgoing :flex-end :flex-start)
         direction (if outgoing :row-reverse :row)]
     (merge message-body-base
-           {:flexDirection direction
+           {:flex-direction direction
             :width         260
```
Comment (flexsurfer) [inline src/status_im/chat/styles/message/message.cljs]: perfectionist crying

**Review by flexsurfer — APPROVED** [review]

**Review by alwx — APPROVED** [review]

File: src/status_im/chat/styles/message/message.cljs (hunk 3 above)
Comment (goranjovic) [reply src/status_im/chat/styles/message/message.cljs]: guilty as charged :)

**Comment by antdanchenko** [comment]
@goranjovic please take a look at `test_send_eth_from_wallet_sign_now ❌ `

**Comment by goranjovic** [comment]
@antdanchenko Does this also appear in develop? Rebasing against it was the only change I made now

**Comment by antdanchenko** [comment]
@goranjovic I am using `im.status.ethereum-d0dff1.apk` nightly build right now and didn't see the issue. But maybe develop branch is already older than the build

**Comment by goranjovic** [comment]
@antdanchenko Either way, it's not supposed to be like that so I'm looking into it.

**Comment by denis-sharypin** [comment]
Hey @goranjovic please pay attention that name of the person should in grey color. Double check in Zeplin

**Comment by annadanchenko** [comment]
Issues:
1. iOS only: username is centered instead of left-aligned
2. For new 1:1 chat no text "There are no messages in this chat yet". Only Today is shown. Expected: no "Today", text "There are no messages  in this chat yet" as in design. When user sends or recieves first message then "Today" should be shown on top of the messages (same style as now)
img_2595
3. ~No Delete chat option for group chat (as supposed to be according to design)~ ignore, PR https://github.com/status-im/status-react/pull/3308 does it
4. ~Settings are shown instead of View Profile for Group chat~ ignore, PR https://github.com/status-im/status-react/pull/3308 does it
img_2600
5. After discussion with @denis-sharypin we need to add Clear History to each menu for Group and Public chat that will clear the content of the chat if tap on it. Same function was available in chat settings before
6. ~For "destructive" actions like leave chat and delete chat: need to have confirmation popup and do it only after confirmed. If not confirmed then keep menu with options opened but do nothing (e.g. do not leave chat). Same style as here is to be used:~ Will add as separate issue
screen shot 2018-02-13 at 19 41 48

**Comment by denis-sharypin** [comment]
Couple more things:

The second line in the navigation bar should be also aligned to the left. Also, mind the gap between
userpick and name.

Message states (Sent, Delivered, Sending…) should have a bit different design. Sorry, we didn't introduce them into Zeplin mocks. It has a text style (margin from the message box, text size) as names except for color. Keep current color for it

slice

**Comment by asemiankevich** [comment]
@goranjovic I am not sure whats wrong but the placeholder is hardly visible (both iOS and Android)  and chat name is not left aligned (iOS)
this is the job i ran to create build (previous one resulted in android null link on diawi https://jenkins.status.im/job/status-react/job/parametrized_build/438/)

also avatars are not shown

image

image

**Comment by goranjovic** [comment]
@asemiankevich I'll take a look. This is really weird, I can only blame gremlins!

**Comment by goranjovic** [comment]
Restricting the scope of this PR to chat messages area only as it's already quite overdue and there is no point rebasing it any further. Creating a new issue for the remaining toolbar discrepancy - here  #3561

**Comment by annadanchenko** [comment]
Ignoring toolbar with chat name and avatar.
issues:
1. Sender's name is above the right side of the message while in design it's above left side
2. Avatar next to the message is shown on the same line as username while in designs it's shown on the same line as the last line of the message 
3. "There are no messages in this chat yet" is left-aligned instead of centred as in design
Now
img_1976
Design:
screen shot 2018-03-15 at 15 29 03

**Comment by annadanchenko** [comment]
agreed to merge current PR and create separate issues for fixing

**Comment by annadanchenko** [comment]
created separate issues
https://github.com/status-im/status-react/issues/3586
https://github.com/status-im/status-react/issues/3587
https://github.com/status-im/status-react/issues/3588

**Pull request merged**